### ISSUE TITLE
Parallel iterator with zipped vectors (arrays)

### ISSUE THREAD

**Issue by fakub** [rayon-rs/rayon#874]
A typical task to do in parallel is to merge two vectors into one. Consider the following code for the addition of respective elements:
```rust
use rayon::prelude::*;
fn main() {
    let mut foo = [1,2,3,4];
    let     bar = [5,6,7,8];
    // nice-to-have: par_iter_mut()     par_iter()
    for (f,b) in foo.iter_mut().zip(bar.iter()) {
        *f += b;
    }
    println!("foo: {:?}", foo);
}
```
This way, works fine. However, the code with `par_iter` and `par_iter`, respectively, does not compile:
```
error[E0277]: `rayon::iter::Zip<rayon::slice::IterMut<'_, {integer}>, rayon::slice::Iter<'_, {integer}>>` is not an iterator
  --> src/main.rs:14:18
   |
 6 |     for (f,b) in foo.par_iter_mut().zip(bar.par_iter()) {
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `rayon::iter::Zip<rayon::slice::IterMut<'_, {integer}>, rayon::slice::Iter<'_, {integer}>>` is not an iterator
   |
   = help: the trait `Iterator` is not implemented for `rayon::iter::Zip<rayon::slice::IterMut<'_, {integer}>, rayon::slice::Iter<'_, {integer}>>`
   = note: required because of the requirements on the impl of `IntoIterator` for `rayon::iter::Zip<rayon::slice::IterMut<'_, {integer}>, rayon::slice::Iter<'_, {integer}>>`
   = note: required by `std::iter::IntoIterator::into_iter`
```
Is this not implemented yet, or is there any other preferred way to merge two vectors into one using parallel iterators? Or, is there any workaround?

**Comment by cuviper** [comment]
The language's `for` loop only works with plain `Iterator`s, but instead you can use `.for_each(|(f, b)| { *f += b });`
